### DISCUSSION TITLE
Recovering my installation / reopening BirdDB.txt

### DISCUSSION THREAD

**Discussion by srchild** [mcguirepr89/BirdNET-Pi#741] [General]
I seem to have messed up a bit...

Following these problems 

https://github.com/mcguirepr89/BirdNET-Pi/discussions/740

https://github.com/mcguirepr89/BirdNET-Pi/discussions/699#discussioncomment-4983499

I reinstalled using:

`cd && uninstall.sh && rm -drfv BirdNET* && curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Pi/main/newinstaller.sh | bash`

Before doing that I made a copy in a safe place of BirdDB.txt

After the reinstall my BirdSongs were intact but BirdDB.txt was an empty file. I smugly copied my backup copy over the empty file and thought that would fix it. But it hasn't.

My installation is running now. But it is not reading the past data in BirdDB.txt. The overview screen shows no detections. The detections from earlier this morning are shown on the overview but no historic data is showing. For example, Recordings and Best Recordings tabs are empty.

Even opening bird.db in adminer show no rows found. This even though the file opens ok in a text editor and shows the expected data.

SELECT "*", "rowid" FROM "detections" LIMIT 50
No rows.

The permissions seems correct

-rw-rw-r-- 1 birduser birduser  5241120 Feb 15 13:59 BirdDB.txt

What am I doing wrong?

Thanks

**Comment by ehpersonal38** [comment]
Hi, in the future make sure you backup `birds.db`, NOT `BirdDB.txt`. But luckily you can convert the latter into the former with these instructions:

https://github.com/mcguirepr89/BirdNET-Pi/discussions/411#discussioncomment-3042528

**Reply by srchild** [reply]
Big whoops! But thanks for the great recovery tip!

I went for the conversion to csv option because that was known to work, whereas Patrick only said that he thought the command line import ought to work but was untested :)

With over 50k rows my csv file was too large for the import by upload. I converted it to SQL INSERT format, uploaded it by sftp named as adminer.sql and put it in the scripts directory. It took fewer than 5 seconds to import all the rows.

Phew!
